xio-core/src/main/java/com/xjeffrose/xio/core/XioIdleDisconnectHandler.java

```diff
@@ -26,6 +27,14 @@ public XioIdleDisconnectHandler(
         TimeUnit.MILLISECONDS);
   }
 
+  @Override
+  public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+    super.userEventTriggered(ctx, evt);
+    if (evt instanceof ProxyClientIdle) {
+      ctx.channel().close();
```

Comment:
This isn’t really in the spirit of what this handler is supposed to be doing. What we probably want is to notify the business logic around the proxy client that things went south.